worker/src/RTC/TransportCongestionControlServer.cpp

```diff
@@ -124,17 +125,77 @@ namespace RTC
 					break;
 				}
 
+				// Only insert the packet when receiving it for the first time.
+				if (!this->mapPacketArrivalTimes.try_emplace(wideSeqNumber, nowMs).second)
+				{
+					break;
+				}
+
+				// We may receive packets with sequence number lower than the one in previous
+				// tcc feedback, these packets may have been reported as lost previously,
+				// therefore we need to reset the start sequence num for the next tcc feedback.
+				if (
+				  !this->transportWideSeqNumberReceived ||
+				  RTC::SeqManager<uint16_t>::IsSeqLowerThan(
+				    wideSeqNumber, this->transportCcFeedbackWideSeqNumStart))
+				{
+					this->transportCcFeedbackWideSeqNumStart = wideSeqNumber;
+				}
+
+				this->transportWideSeqNumberReceived = true;
+
+				MayDropOldPacketArrivalTimes(wideSeqNumber, nowMs);
+
 				// Update the RTCP media SSRC of the ongoing Transport-CC Feedback packet.
 				this->transportCcFeedbackSenderSsrc = 0u;
 				this->transportCcFeedbackMediaSsrc  = packet->GetSsrc();
 
-				this->transportCcFeedbackPacket->SetSenderSsrc(0u);
-				this->transportCcFeedbackPacket->SetMediaSsrc(this->transportCcFeedbackMediaSsrc);
+				MaySendLimitationRembFeedback(nowMs);
+
+				break;
+			}
+
+			case RTC::BweType::REMB:
+			{
+				uint32_t absSendTime;
+
+				if (!packet->ReadAbsSendTime(absSendTime))
+				{
+					break;
+				}
+
+				// NOTE: nowMs is uint64_t but we need to "convert" it to int64_t before
+				// we give it to libwebrtc lib (althought this is implicit in the
+				// conversion so it would be converted within the method call).
+				auto nowMsInt64 = static_cast<int64_t>(nowMs);
+
+				this->rembServer->IncomingPacket(nowMsInt64, packet->GetPayloadLength(), *packet, absSendTime);
+
+				break;
+			}
+		}
+	}
+
+	void TransportCongestionControlServer::FillAndSendTransportCcFeedback()
+	{
+		MS_TRACE();
+
+		if (!this->transportWideSeqNumberReceived)
+		{
+			return;
+		}
+
+		auto it = this->mapPacketArrivalTimes.lower_bound(this->transportCcFeedbackWideSeqNumStart);
 
-				// Provide the feedback packet with the RTP packet info. If it fails,
-				// send current feedback and add the packet info to a new one.
+		if (it != this->mapPacketArrivalTimes.end())
```

Comment:
Here please let's do early return if the condition is not satisfied (instead of wrapping all code into a condition block):

```c++
if (it == this->mapPacketArrivalTimes.end())
{
  return;
}

// Rest of the code here.
```